[Logs UI] Show log analysis ML jobs in a list

weltenwort opened the pull request:
## Summary

This modifies the ML job setup flyout of the anomalies tab to offer a list of the two available modules. Via the list each of the modules' jobs can be created or re-created.

closes #64757 

## Previews

![image](https://user-images.githubusercontent.com/973741/87173978-ce690380-c2d6-11ea-9cef-87358d991282.png)

![image](https://user-images.githubusercontent.com/973741/87173280-ca88b180-c2d5-11ea-910d-3d883b71db5e.png)

![image](https://user-images.githubusercontent.com/973741/87173326-dc6a5480-c2d5-11ea-9b88-9f955457bdc1.png)

## Review and test notes

- The implementation can be seen as rather simplistic because it hard-codes the two available modules in the results page and flyout. The advantage is that the changeset is significantly smaller than of a fully generic implementation.
- The possible setup state were reduced because the visibility state of the flyout and its pages is tracked in a separate flyout state (shared via the context).
- The category page flyout only contains the category setup form and should function the same as before.
- Warnings for outdated configurations or definitions can be triggered by modifying the jobs with queries like the following:

  ```
  POST _ml/anomaly_detectors/kibana-logs-ui-${spaceId}-default-log-entry-categories-count/_update
  {
    "custom_settings": {
      "created_by": "ml-module-logs-ui-categories",
      "job_revision": 0, // decrement this to cause definition warning
      "logs_source_config": {
        "indexPattern": "filebeat-*", // add something here to cause configuration warning
        "timestampField": "@timestamp",
        "bucketSpan": 900000,
        "datasetFilter": {
          "type": "includeAll"
        }
      }
    }
  }
  ```

  ```
  POST _ml/anomaly_detectors/kibana-logs-ui-${spaceId}-default-log-entry-rate/_update
  {
    "custom_settings": {
      "created_by": "ml-module-logs-ui-analysis",
      "job_revision": 2, // decrement this to cause definition warning
      "logs_source_config": {
        "indexPattern": "filebeat-*", // add something here to cause configuration warning
        "timestampField": "@timestamp",
        "bucketSpan": 900000
      }
    }
  }
  ```

## Future work

- Make module management and setup hooks multi-module-aware.
- Combine the module state warning callouts into a combined callout (possible per module).
- Show the module state warnings in the flyout and the setup forms.
- Move the setup states to a provider so they survive closing the flyout.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/)
* Commit: 9f460539a174745fe032653e472537bb20fc447f
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/execution/node/312/log)

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/9f460539a174745fe032653e472537bb20fc447f) | diff | [baseline](https://github.com/elastic/kibana/commit/005128c22306dea4eb36b6301f49f9bd03015602) |
| --- | --- | --- | --- |
| infra | 599 | -7 | 606 |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/","number":"60624","commit":"9f460539a174745fe032653e472537bb20fc447f"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/)
* Commit: 564acf74c9890ee265cfd6bc983b1cafd4fc49e6

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/564acf74c9890ee265cfd6bc983b1cafd4fc49e6) | diff | [baseline](https://github.com/elastic/kibana/commit/005128c22306dea4eb36b6301f49f9bd03015602) |
| --- | --- | --- | --- |
| infra | 599 | -7 | 606 |

### History
* :broken_heart: [Build #60624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/) failed 9f460539a174745fe032653e472537bb20fc447f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/","number":"60624","commit":"9f460539a174745fe032653e472537bb20fc447f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/","number":"60796","commit":"564acf74c9890ee265cfd6bc983b1cafd4fc49e6"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/)
* Commit: 0a5557370d80fa5f6d3159f6690e1071d2374128
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/execution/node/308/log)

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/0a5557370d80fa5f6d3159f6690e1071d2374128) | diff | [baseline](https://github.com/elastic/kibana/commit/cd2d1b8053da9ca37b44fc3edc9c83bd72eecc46) |
| --- | --- | --- | --- |
| infra | 599 | -7 | 606 |

### History
* :green_heart: [Build #60796](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/) succeeded 564acf74c9890ee265cfd6bc983b1cafd4fc49e6
* :broken_heart: [Build #60624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/) failed 9f460539a174745fe032653e472537bb20fc447f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/","number":"60624","commit":"9f460539a174745fe032653e472537bb20fc447f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/","number":"60796","commit":"564acf74c9890ee265cfd6bc983b1cafd4fc49e6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/","number":"60864","commit":"0a5557370d80fa5f6d3159f6690e1071d2374128"}]}
PIPELINE-->

Kerry350 commented on x-pack/plugins/infra/public/pages/logs/page_content.tsx:
```diff
@@ -42,10 +42,10 @@ export const LogsPageContent: React.FunctionComponent = () => {
     app: 'logs',
-    title: logRateTabTitle,
-    pathname: '/log-rate',
+    title: anomaliesTabTitle,
```
🎉🎉🎉🎉🎉

weltenwort commented on x-pack/plugins/infra/public/pages/logs/log_entry_rate/page_content.tsx:
```diff
@@ -65,31 +100,42 @@ export const LogEntryRatePageContent = () => {
-        <LogEntryRateSetupFlyout isOpen={isFlyoutOpen} onClose={closeFlyout} />
+        <LogEntryRateResultsContent />
+        <LogAnalysisSetupFlyout />
+        {/* <LogEntryRateSetupFlyout isOpen={isFlyoutOpen} onClose={closeFlyout} /> */}
```
oops, of course :grimacing:

weltenwort commented on x-pack/plugins/infra/public/components/logging/log_analysis_setup/setup_flyout/setup_flyout.tsx:
```diff
@@ -0,0 +1,80 @@
+      <EuiButtonEmpty flush="left" iconSide="left" iconType="arrowLeft" onClick={onViewModuleList}>
+        <FormattedMessage
+          id="xpack.infra.logs.analysis.setupFlyoutGotoListButtonLabel"
+          defaultMessage="All Machine Learning jobs"
```
Good question, I'm trying to find that out...

weltenwort replied:
Sorry that it took a while to find an answer. The preferred variant is to spell out "Machine Learning" (capitalized if it refers to the app) and only use the abbreviation if there's too little room.

weltenwort commented:
Thank you for the timely review. I mostly addressed your comments, but I'm still looking for writer input regarding the ML vs Machine Learning question.

> Only thing to mention out of these screenshots is the "success" callout rendered for me with the log entry job, should it do the same for just the categories job?

That condition could indeed be better. I just changed it to "when any of the jobs have just been created and no log rate or anomalies are found". I imagine that would be the situation in which the user would be most unsure. Does that make sense?

elasticmachine commented:
Pinging @elastic/logs-metrics-ui (Team:logs-metrics-ui)

weltenwort commented:
:information_source: I rebased this on `master` because #70982 has been merged.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60976/)
* Commit: 9eb33f1472d05d0cda6224dd82f0abe404d6bb2d

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/9eb33f1472d05d0cda6224dd82f0abe404d6bb2d) | diff | [baseline](https://github.com/elastic/kibana/commit/cd2d1b8053da9ca37b44fc3edc9c83bd72eecc46) |
| --- | --- | --- | --- |
| infra | 599 | -7 | 606 |

### History
* :broken_heart: [Build #60864](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/) failed 0a5557370d80fa5f6d3159f6690e1071d2374128
* :green_heart: [Build #60796](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/) succeeded 564acf74c9890ee265cfd6bc983b1cafd4fc49e6
* :broken_heart: [Build #60624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/) failed 9f460539a174745fe032653e472537bb20fc447f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/","number":"60624","commit":"9f460539a174745fe032653e472537bb20fc447f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/","number":"60796","commit":"564acf74c9890ee265cfd6bc983b1cafd4fc49e6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/","number":"60864","commit":"0a5557370d80fa5f6d3159f6690e1071d2374128"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60976/","number":"60976","commit":"9eb33f1472d05d0cda6224dd82f0abe404d6bb2d"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60986/)
* Commit: 3bbf3e2d80fb1c8502a112e2a28cbff1c2d80a2c

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/3bbf3e2d80fb1c8502a112e2a28cbff1c2d80a2c) | diff | [baseline](https://github.com/elastic/kibana/commit/cd2d1b8053da9ca37b44fc3edc9c83bd72eecc46) |
| --- | --- | --- | --- |
| infra | 599 | -7 | 606 |

### History
* :green_heart: [Build #60976](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60976/) succeeded 9eb33f1472d05d0cda6224dd82f0abe404d6bb2d
* :broken_heart: [Build #60864](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/) failed 0a5557370d80fa5f6d3159f6690e1071d2374128
* :green_heart: [Build #60796](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/) succeeded 564acf74c9890ee265cfd6bc983b1cafd4fc49e6
* :broken_heart: [Build #60624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/) failed 9f460539a174745fe032653e472537bb20fc447f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/","number":"60624","commit":"9f460539a174745fe032653e472537bb20fc447f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/","number":"60796","commit":"564acf74c9890ee265cfd6bc983b1cafd4fc49e6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/","number":"60864","commit":"0a5557370d80fa5f6d3159f6690e1071d2374128"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60976/","number":"60976","commit":"9eb33f1472d05d0cda6224dd82f0abe404d6bb2d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60986/","number":"60986","commit":"3bbf3e2d80fb1c8502a112e2a28cbff1c2d80a2c"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61026/)
* Commit: e412decf69097ffc4dc6442052a9edf14990800b

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/e412decf69097ffc4dc6442052a9edf14990800b) | diff | [baseline](https://github.com/elastic/kibana/commit/41c4f18b8961dcfe537c727c8547d28de7c8c501) |
| --- | --- | --- | --- |
| infra | 599 | -8 | 607 |

### History
* :green_heart: [Build #60986](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60986/) succeeded 3bbf3e2d80fb1c8502a112e2a28cbff1c2d80a2c
* :green_heart: [Build #60976](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60976/) succeeded 9eb33f1472d05d0cda6224dd82f0abe404d6bb2d
* :broken_heart: [Build #60864](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/) failed 0a5557370d80fa5f6d3159f6690e1071d2374128
* :green_heart: [Build #60796](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/) succeeded 564acf74c9890ee265cfd6bc983b1cafd4fc49e6
* :broken_heart: [Build #60624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/) failed 9f460539a174745fe032653e472537bb20fc447f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60624/","number":"60624","commit":"9f460539a174745fe032653e472537bb20fc447f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60796/","number":"60796","commit":"564acf74c9890ee265cfd6bc983b1cafd4fc49e6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60864/","number":"60864","commit":"0a5557370d80fa5f6d3159f6690e1071d2374128"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60976/","number":"60976","commit":"9eb33f1472d05d0cda6224dd82f0abe404d6bb2d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/60986/","number":"60986","commit":"3bbf3e2d80fb1c8502a112e2a28cbff1c2d80a2c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61026/","number":"61026","commit":"e412decf69097ffc4dc6442052a9edf14990800b"}]}
PIPELINE-->

weltenwort commented:
I'm still waiting for new insights on the copy, but I'll merge while the Jenkins and GitHub stars are aligned.
